225 235 Add madeBySystem and madeExecution

dr-shorthair opened the pull request:
to sosa-common, and sub-property axioms to ssn-* modules

Add usedActuatingProcedure usedObservingProcedure usedSamplingProcedure to sosa-* modules, and  sub-property axioms to ssn-* modules 

Add actsOn to sosa-actuation

Closes #225

dr-shorthair committed:
Add madeBySystem and madeExecution to sosa-common, and sub-property axioms to ssn-* modules
Add usedActuatingProcedure usedObservingProcedure usedSamplingProcedure to sosa-* modules, and  sub-property axioms to ssn-* modules
Add actsOn to sosa-actuation

dr-shorthair committed:
Update diagrams

dr-shorthair committed:
ChangeLog
Fix diagrams

dr-shorthair commented:
Terms in red are the proposed additions, classes in brown are changes we already introduced

![System-Execution-Procedure-specializations](https://github.com/w3c/sdw-sosa-ssn/assets/608303/773bea5e-7c6b-4f5c-b173-7ec4ce58ab21)

ldesousa commented:
I don't see technical issues with this implementation (yet), but I would like to think a bit more on this from higher level of abstraction. The ontology is moving towards a set of four very similar graphs, centered on the four main classes:
- Execution
- Actuation
- Observation
- Sampling

Those classes are well defined and understood, not only by those of us in the group, but also to users in general that may deal with concerned datasets in this context. However, the semantic nuances between the sub-classes of Procedure and Property are less evident, i.e.:
- ActuatingProcedure
- ObservingProcedure
- SamplingProcedure
- ActuableProperty
- ObservableProperty
- ~~SamplingProperty~~

Are all those classes necessary? Do they all have a well defined and unique semantic definition?

Finally coming to the predicates: are they all semantically unique? Or is a single predicate object property between Execution and Procedure enough to convey that relation in all sub-graphs? I would rather have a leaner ontology, with as many elements as possible at the Execution (abstract?) level, only to be specialised in case of objective semantic differences. 

Thoughts?

dr-shorthair commented:
Thanks @ldesousa - this is exactly the conversation we need. Please continue over in #225

dr-shorthair committed:
Merge branch 'gh-pages' into 225-madebysystem

# Conflicts:
#	ssn/index.html

dr-shorthair committed:
Merge branch 'gh-pages' into 225-madebysystem

# Conflicts:
#	ssn/images/Actuation.png
#	ssn/images/Observation.png
#	ssn/images/System.png
#	ssn/images/System.svg
#	ssn/index.html
#	ssn/rdf/ontology/core/sosa-actuation.ttl

dr-shorthair committed:
Tidy up documentation

dr-shorthair committed:
Remove (comment out) usedProcedure specializations

oldskeptic commented:
> Now we have introduced a generalization `Execution`, and specializations `ActuatingProcedure`, `ObservingProcedure`, `SamplingProcedure` (note that `ActuatableProperty` and `ObservableProperty` were there before). The proposed predicates (in red above) are really just my chasing through the consequences.

Do we need the specializations `ActuatingProcedure`, `ObservingProcedure`, `SamplingProcedure`? Is there a material difference in these classes that wouldn't be covered by `Procedure`? A bit late for this question I realise.

dr-shorthair committed:
Merge branch 'gh-pages' into 225-madebysystem

# Conflicts:
#	ssn/chapters/Actuation.html
#	ssn/chapters/Common.html
#	ssn/chapters/Observation.html
#	ssn/chapters/Sampling.html
#	ssn/images/Actuation.png
#	ssn/images/Execution.png
#	ssn/images/Observation.png
#	ssn/images/System.png
#	ssn/index.html
#	ssn/rdf/ontology/core/sosa-actuation.ttl
#	ssn/rdf/ontology/core/sosa-common.ttl
#	ssn/rdf/ontology/core/sosa-observation.ttl
#	ssn/rdf/ontology/core/sosa-sampling.ttl
#	ssn/rdf/ontology/core/sosa.ttl
#	ssn/rdf/ontology/core/ssn-actuation.ttl
#	ssn/rdf/ontology/core/ssn-observation.ttl
#	ssn/rdf/ontology/core/ssn-sampling.ttl

dr-shorthair commented:
I've copied this discussion back to issue #225 - better to have it there rather than on the PR. 
Please lets keep the high-level discussion there. 
PR comments should only be on whether the PR addresses the agreed resolution.

dr-shorthair committed:
Moved ObservingProcedure and Sampling Procedure back to sosa-oms: namespace
Additional text on Execution and System to emphasize that they are generalized/abstract classes and that individuals in applications should use sub-classes.
New diagram in overview establishing the generalized Execution pattern.

dr-shorthair committed:
Remove specialized procedures from diagrams

dr-shorthair commented:
Preview https://raw.githack.com/w3c/sdw-sosa-ssn/225-madebysystem/ssn/index.html

Closes #225 
Closes #235

dr-shorthair committed:
Includes madeBySystem/madeExecution sub-property relationships
Add notes about non-use of abstract properties

rob-metalinkage reviewed (dismissed):
I think the text clarification should be emphasised as SHOULD NOT

dr-shorthair committed:
Render key words in ALL-CAPS

dr-shorthair commented:
@rob-metalinkage - fixed

dr-shorthair committed:
madeBySystem and madeExecution missing from two azlists

dr-shorthair committed:
Correct changelog

ldesousa commented on ssn/rdf/ontology/core/sosa-common.ttl:
```diff
@@ -229,7 +229,14 @@ sosa:Execution
+  It appears in the ontology to support the description of a common pattern for executions and their properties. 
+
+  Execution may be considered to be abstract, so individuals SHOULD NOT be declared to be members of only this class. 
+  Applications SHOULD use members of concrete sub-classes of Execution.
```
The term "member" is not that common in the Semantic Web. I suggest being more precise here, using instead "object properties". Besides, the most important sentence is the second, this third sentence could even be absent.

dr-shorthair replied:
I don't understand where 'object properties' would appear in the text. 

I used 'members' in the sense of the 'class extension'. 
Every individual of type `ex:Class1` is a _member_ of that class. 

I agree that the two SHOULD sentences are redundant. 
Perhaps it would be better phrased as 
> Execution MUST be considered to be abstract, so any individual Execution SHOULD be declared to be an instance of one of the concrete sub-classes, and not of the Execution class.

ldesousa commented on ssn/rdf/ontology/core/sosa-common.ttl:
```diff
@@ -341,6 +348,24 @@ sosa:isResultOf
+  madeBySystem generalizes the madeByActuator, madeBySensor, and madeBySampler properties. 
+  It appears in the ontology to support the description of a common pattern for executions and their properties. 
+
+  madeBySystem may be considered to be abstract, so SHOULD NOT appear in data instances. 
```
"may be" and "SHOULD NOT" possibly not aligned. I would rather have "should be considered" or even "must be considered".

dr-shorthair replied:
> madeBySystem MUST be considered to be abstract, so SHOULD NOT appear in data instances.
> Applications can use concrete sub-properties of madeBySystem.

ldesousa commented on ssn/rdf/ontology/core/sosa-common.ttl:
```diff
@@ -371,11 +396,11 @@ sosa:phenomenonTime
   The phenomenonTime is not necessarily the same as the resultTime. 
   
-  In the case of Actuation, phenomenonTime should indicate the time that the actuator was active. 
+  In the case of Actuation, phenomenonTime SHOULD indicate the time that the actuator was active. 
```
"the time at which the actuator was active"

dr-shorthair replied:
OK

ldesousa commented on ssn/rdf/ontology/core/sosa-common.ttl:
```diff
@@ -371,11 +396,11 @@ sosa:phenomenonTime
   In the case of Observation, the phenomenonTime may be distant from the resultTime, for example in the case of forecasts the phenomenonTime is after the resultTime, and in historical, geological and cosmological investigations the phenomenonTime may be far in the past while the resultTime is contemporary. 
 
-  In the case of Sampling the phenomenonTime should indicate the period during which the Sampling Procedure was active, while the resultTime may indicate when the sample comes into the posession of the operator. 
+  In the case of Sampling the phenomenonTime SHOULD indicate the period during which the Sampling Procedure was active, while the resultTime may indicate when the sample comes into the posession of the operator. 
```
This sentence is too long, imho. I suggest this alternative formulation:
"In the case of Sampling, phenomenonTime SHOULD indicate the period during which the Sampling Procedure was active. Whereas resultTime may indicate when the sample comes into the possession of the operator."

dr-shorthair replied:
Sentences broken up:
> In the case of Observation, the phenomenonTime MAY be distant from the resultTime. 
>  For example in the case of forecasts the phenomenonTime is after the resultTime. 
>  In historical, geological and cosmological investigations the phenomenonTime can be far in the past, while the resultTime is contemporary. 
>
>  In the case of Sampling the phenomenonTime SHOULD indicate the time during which the Sampling Procedure was active.
>  The resultTime can indicate when the sample comes into the posession of the operator.

ldesousa commented on ssn/rdf/ontology/core/sosa-common.ttl:
```diff
@@ -443,8 +468,16 @@ sosa:System
+  System generalizes the Actuator, Sensor, and Sampler classes. 
+  It appears in the ontology to support the description of a common pattern for executions and their properties. 
+
+  System may be considered to be abstract, so individuals SHOULD NOT be declared to be members of only this class. 
```
"instances" instead of "members". That is more common in the Semantic Web.

dr-shorthair replied:
Re-worded consistent with `Execution`

>  System MUST be considered to be abstract, so any individual System SHOULD be declared to be an instance of one of the concrete sub-classes, and not of the System class.

ldesousa commented on ssn/rdf/ontology/core/sosa-common.ttl:
```diff
@@ -561,6 +594,24 @@ sosa:isSubSystemOf
+  madeExecution generalizes the madeActuation, madeObservation, and madeSampling properties. 
+  It appears in the ontology to support the description of a common pattern for executions and their properties. 
+
+  madeExecution may be considered to be abstract, so SHOULD NOT appear in data instances. 
```
As before, I suggest replacing "may be" for "should be" or "must be". The architect(s) has a clear intention here.

dr-shorthair replied:
>   madeExecution MUST be considered to be abstract, so SHOULD NOT appear in data instances. 
>  Applications can use concrete sub-properties of madeExecution.

ldesousa commented on ssn/rdf/ontology/core/sosa.ttl:
```diff
@@ -31,6 +31,24 @@ sosa:
   owl:imports sosa-act: , sosa-obs: , sosa-sam:.
 
+sosa:madeBySystem 
+  schema:domainIncludes sosa:Actuation ;
```
In this cases it is more common to see the use of a list. E.g.:
`schema:domainIncludes [sosa:Actuation, sosa:Observation, sosa:Sampling] ;`
Makes it easier to read, in my view.

dr-shorthair replied:
No `[  ]` required. In TTL a set of objects for the same predicate is simply comma-separated. 

However, readability is not the key concern in the RDF/TTL. 
They are broken onto separate lines for ease of editing/bookkeeping at this stage. 
The RDF/TTL can be re-styled later if desired.

ldesousa commented on ssn/rdf/ontology/core/sosa.ttl:
```diff
@@ -31,6 +31,24 @@ sosa:
+  rdfs:isDefinedBy sosa-common: .
+
+sosa:madeExecution 
+  schema:domainIncludes sosa:Actuator ;
```
Ditto.

dr-shorthair replied:
Ditto.

ldesousa commented on ssn/images/Execution.svg:
Would it be simple to convey here the different namespaces (SOSA, SOSA-OMS)? Say with graphical UML elements like packages.

dr-shorthair replied:
Ignore the SVG - they are not up-to-date. 
I'll remove them (for now).

ldesousa commented on ssn/chapters/Common.html:
```diff
@@ -1186,20 +1145,27 @@ <h6><a href="#SOSAExecution">sosa:Execution</a></h6>
+                                It appears in the ontology to support the description of a common pattern for executions
+                                and their properties.
+
+                                Execution may be considered to be abstract, so individuals SHOULD NOT be declared to be
```
Consider replacing "may" with "should" or "must". See comments on the TTL.

dr-shorthair replied:
I've made the HTML consistent with RDF/TTL everywhere.

ldesousa commented on ssn/chapters/Common.html:
```diff
@@ -1186,20 +1145,27 @@ <h6><a href="#SOSAExecution">sosa:Execution</a></h6>
+
+                                Execution may be considered to be abstract, so individuals SHOULD NOT be declared to be
+                                members of only this class.
+                                Applications SHOULD use members of concrete sub-classes of Execution.
```
"instances" instead of "members". Consistent with comments on the TTL.

dr-shorthair replied:
Yes.

ldesousa commented on ssn/chapters/Common.html:
```diff
@@ -1769,6 +1735,93 @@ <h6><a href="#SOSAisResultOf">sosa:isResultOf</a></h6>
+                            It appears in the ontology to support the description of a common pattern for executions and
+                            their properties.
+
+                            <code>madeBySystem</code> may be considered to be abstract, so SHOULD NOT appear in data
```
"should" or "must" in place of "may".

dr-shorthair replied:
Yes

ldesousa commented on ssn/chapters/Common.html:
```diff
@@ -1903,7 +1956,7 @@ <h5><a href="#SOSAphenomenonTime">sosa:phenomenonTime</a></h5>
                 <br>
                 In the case of <a href="#SOSAActuation">sosa:Actuation</a>, <a
-                    href="#SOSAphenomenonTime">sosa:phenomenonTime</a> should indicate the time that the <a
+                    href="#SOSAphenomenonTime">sosa:phenomenonTime</a> SHOULD indicate the time that the <a
```
"the time at which the sosa:Actuator was active"

dr-shorthair replied:
Yes

ldesousa commented on ssn/chapters/Common.html:
```diff
@@ -2389,11 +2437,27 @@ <h6><a href="#SOSASystem">sosa:System</a></h6>
+                                It appears in the ontology to support the description of a common pattern for executions
+                                and their properties.
+
+                                System may be considered to be abstract, so individuals SHOULD NOT be declared to be
```
"must" or "should" in place of "may"

dr-shorthair replied:
Yes

ldesousa commented on ssn/chapters/Common.html:
```diff
@@ -2973,6 +3037,86 @@ <h6><a href="#SOSAisSubSystemOf">sosa:isSubSystemOf</a></h6>
+                            It appears in the ontology to support the description of a common pattern for executions and
+                            their properties.
+
+                            <code>madeExecution</code> may be considered to be abstract, so SHOULD NOT appear in data
```
"must" or "should" in place of "may"

dr-shorthair replied:
Yes

ldesousa commented on ssn/images/System.svg:
There is a problem with this file. Both Gimp and Inkscape complain it has errors. GitHub can't render it.

oldskeptic replied:
I'm still having trouble viewing this file again.

dr-shorthair replied:
Ignore SVG versions. Removed now.

ldesousa committed:
Corrects small typos

KathiSchleidt commented on ssn/images/Execution.svg:
Still has specialized Procedure Types

dr-shorthair replied:
SVG all obsolete. Ignore (removed).

dr-shorthair committed:
Remove SVG versions of diagrams (for now)

dr-shorthair committed:
Changes in response to @ldesousa review

dr-shorthair committed:
Don't need non-normative warnings in OMS alignment section - everything is new here

dr-shorthair committed:
Fix non-normative warnings conistent with @ldesousa typo fixes

dr-shorthair committed:
Merge remote-tracking branch 'origin/225-madebysystem' into 225-madebysystem

# Conflicts:
#	ssn/chapters/Common.html
#	ssn/chapters/OMS-alignment.html

dr-shorthair committed:
Add note about why new terms are currently non-normative
